pkg/image/apis/image/helper.go

```diff
@@ -397,13 +397,25 @@ func FollowTagReference(stream *ImageStream, tag string) (finalTag string, ref *
 			// no tag at the end of the rainbow
 			return tag, nil, false, multiple
 		}
-		if tagRef.From == nil || tagRef.From.Kind != "ImageStreamTag" || strings.Contains(tagRef.From.Name, ":") {
+		if tagRef.From == nil || tagRef.From.Kind != "ImageStreamTag" {
 			// terminating tag
 			return tag, &tagRef, true, multiple
 		}
 
-		// follow the referenec
-		tag = tagRef.From.Name
+		// The reference needs to be followed with two format patterns:
+		// a) sameis:sometag and b) sometag
+		if strings.Contains(tagRef.From.Name, ":") {
+			parts := strings.Split(tagRef.From.Name, ":")
+			if parts[0] != stream.ObjectMeta.Name {
+				// anotheris:sometag - this should not happen.
+				return tag, nil, true, multiple
```

Comment:
seems like "ok" should be false here.

Reply:
Actually I also wanted to return "ok" with false, but `FollowTagReference` already has another conditions to returns `tag, nil, false, multiple` as:

https://github.com/openshift/origin/blob/607caee016274b5a352ee9739886cea9e497d87a/pkg/image/apis/image/helper.go#L390-L391
https://github.com/openshift/origin/blob/607caee016274b5a352ee9739886cea9e497d87a/pkg/image/apis/image/helper.go#L397-L398

So, if here (cross-imagestream error) also returns ok with false in the same way, `importimage.go` cannot distinguish and handle it as cross-imagestream error. 
(As another approach, I considered to add one more return value as well, but ok with true is better to add extra return val I think.)

Reply:
yes i think this function needs to be refactored to return an error instead of just "ok/not-ok", that way callers can distinguish/report on why the tag could not be followed.

Reply:
Alright, updated to return an error.